src/graphql/db/index.ts

```diff
@@ -0,0 +1,43 @@
+import { User } from "../../schema"
+import { NotFoundError } from "../../error"
+import { baseLogger } from "../../logger"
+import { caseInsensitiveRegex } from "src/utils"
+
+const logger = baseLogger.child({ module: "admin" })
+
+export async function usernameExists({ username }): Promise<boolean> {
+  return Boolean(await User.findOne({ username: caseInsensitiveRegex(username) }))
+}
+
+export const updateUserLevel = async ({ uid, level }) => {
```

Comment:
The existing function has arg validation [here](https://github.com/GaloyMoney/galoy/blob/main/src/AdminOps.ts#L13):
```
  if (Levels.indexOf(level) === -1) {
    const error = `${level} is not a valid user level`
    throw new ValidationError(error, { forwardToClient: true, logger, level: "warn" })
  }
```

Reply:
- I removed `findOrCreate` and used the code directly in the tests for now. I don't like that I had to use that method but the tests were giving me too many troubles otherwise. I'll pair with Juan on making this better moving forward.

- I moved User db methods under `src/db/user.ts` or now. I look at this db directory as the entry point for all the db processes. Let's talk about this a bit more in the next code review.

- The levels check, I left that off as the only place where this function is used is in the GraphQL mutation (and the validation is done on that level). I'm not sure if we have any database-level validation (here and elsewhere) but If we don't, let's start talking about that and how to match these validations in the codebase where needed.